crates/compiler/solve/src/solve.rs

```diff
@@ -1245,39 +1261,45 @@ fn solve(
 
                     state
                 } else {
-                    // work in the next pool to localize header
-                    let next_rank = rank.next();
-
-                    // introduce variables
-                    for &var in rigid_vars.iter().chain(flex_vars.iter()) {
-                        subs.set_rank(var, next_rank);
-                    }
+                    // If the let-binding is generalizable, work at the next rank (which will be
+                    // the rank promoted to generalization); otherwise, stay at the current level.
```

Comment:
is a rank "promoted to generalization"? maybe this phrasing can be clearer. "for which we will attempt generalization" maybe?